src/model_repository_manager/model_lifecycle.cc

```diff
@@ -799,10 +801,20 @@ ModelLifeCycle::OnLoadFinal(
     // Mark current versions ready and track info in foreground
     for (auto& loaded : load_tracker->load_set_) {
       std::lock_guard<std::mutex> curr_info_lk(loaded.second->mtx_);
-
       loaded.second->state_ = ModelReadyState::READY;
       loaded.second->state_reason_.clear();
-
+#ifdef TRITON_ENABLE_METRICS
+      auto reporter = loaded.second->model_->MetricReporter();
+      const uint64_t now_ns =
+          std::chrono::duration_cast<std::chrono::nanoseconds>(
+              std::chrono::steady_clock::now().time_since_epoch())
+              .count();
+      uint64_t time_to_load_ns = now_ns - loaded.second->load_start_ns_;
+      std::chrono::duration<double> time_to_load =
+          std::chrono::duration_cast<std::chrono::duration<double>>(
+              std::chrono::nanoseconds(time_to_load_ns));
```

Comment:
Why don't you time the load time [here](https://github.com/triton-inference-server/core/blob/main/src/model_repository_manager/model_lifecycle.cc#L530) and directly storing the load time in model info. You can still put the metric update logic here.

Reply:
Reason I added in the metric update in 
`ModelLifeCycle::OnLoadFinal()`  was because that is where we set the status to READY for the model
`loaded.second->state_ = ModelReadyState::READY;`
If I set the model load time [here](https://github.com/triton-inference-server/core/blob/main/src/model_repository_manager/model_lifecycle.cc#L530) there is chance that `ModelLifeCycle::OnLoadFinal()` exits without loading the model and we might have a incorrect metric.